### ISSUE TITLE
[Feature] Specify port via cli options

### ISSUE THREAD

**Issue by mkhodan** [microsoft/playwright#14496]
We are using NX for our monorepo and are trying to switch from Protractor to Playwright. We have a custom dev server which launches application and allows us to run several test suites in parallel. The problem is that it uses a dynamic port for each parallel run therefore I need to somehow specify it for playwright. I can do it in config file but this way it is static. I would like to have a cli option or some other way to specify port.

Update: seems like I was looking onto wrong part of config as I'm only able to specify port for webserver which is not what I'm looking for, I would like to specify baseURL via CLI.

**Comment by mkhodan** [comment]
@aslushnikov not really, it would allow me to specify one port but in case of running more than 1 instance I can't differentiate them in playwright config. So I would like to end up with something like all 3 running in parallel:
```
npx playwright test --config=${configPath} --baseURL=http://localhost:63001
npx playwright test --config=${configPath} --baseURL=http://localhost:63002
npx playwright test --config=${configPath} --baseURL=http://localhost:63003
```

where port in each of them is generated by same dev server that runs these commands.

**Comment by aslushnikov** [comment]
@mkhodan hmm, I don't think I follow, (but I also have never had any experience with NX).

> I would like to specify baseURL via CLI.

How's CLI better / different from env variable? If this works for you:

```bash
npx playwright test --config=${configPath} --baseURL=http://localhost:63001
npx playwright test --config=${configPath} --baseURL=http://localhost:63002
npx playwright test --config=${configPath} --baseURL=http://localhost:63003
```
then the following will also work for you:

```bash
BASE_URL=http://localhost:63001 npx playwright test --config=${configPath}
BASE_URL=http://localhost:63002 npx playwright test --config=${configPath} 
BASE_URL=http://localhost:63003 npx playwright test --config=${configPath}
```

**Comment by mkhodan** [comment]
@aslushnikov at first I thought that if all 3 commands are ran simultaneously(in async manner) I could end up in a situation where env variable is updated faster than playwright is able to read it(and as a result tests are ran on wrong url) but it doesn't seem to be an issue after several tries.

Thank you for your idea, it works for me!

Anyway I still think that ability to provide baseURL as cli option is a nice to have feature, though it is not a priority for me anymore.

Cheers :)

**Comment by aslushnikov** [comment]
> Thank you for your idea, it works for me!

Yay!
